server/src/main/java/org/opensearch/gateway/ReplicaShardBatchAllocator.java

```diff
@@ -168,6 +168,31 @@ public void allocateUnassignedBatch(List<ShardRouting> shardRoutings, RoutingAll
         logger.trace("Finished shard allocation execution for unassigned replica shards: {}", shardRoutings.size());
     }
 
+    protected void allocateUnassignedBatchOnTimeout(List<ShardRouting> shardRoutings, RoutingAllocation allocation) {
+        Set<ShardId> shardIdsFromBatch = new HashSet<>();
+        for (ShardRouting shardRouting : shardRoutings) {
+            ShardId shardId = shardRouting.shardId();
+            shardIdsFromBatch.add(shardId);
+        }
+        RoutingNodes.UnassignedShards.UnassignedIterator iterator = allocation.routingNodes().unassigned().iterator();
+        while (iterator.hasNext()) {
+            ShardRouting unassignedShard = iterator.next();
+            AllocateUnassignedDecision allocationDecision;
+            if (!unassignedShard.primary() && shardIdsFromBatch.contains(unassignedShard.shardId())) {
+                allocationDecision = new AllocateUnassignedDecision(
+                    UnassignedInfo.AllocationStatus.DECIDERS_THROTTLED,
+                    null,
+                    null,
+                    null,
+                    false,
+                    0L,
+                    0L
+                );
+                executeDecision(unassignedShard, allocationDecision, allocation, iterator);
+            }
```

Comment:
code duplication between primary/ replica

Reply:
Updated and moved to BaseGatewayShardAllocator